.github/workflows/build-cloud-img.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and push cloud image
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+    - development
+    tags:
+      - v*
+
+permissions:
+  contents: read
+  packages: write
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  docker:
+    strategy:
+      fail-fast: false
+      matrix:
+        # only x86_64 for now. aarch64+cuda isn't really a thing yet
+        arch:
+        - x86_64
+    runs-on: ubuntu-latest
+    name: ${{ matrix.arch }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        tags: |
+          type=ref,event=branch
+          type=ref,event=tag
+          type=ref,event=pr
+          type=sha
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - if: github.event_name != 'pull_request'
+      name: Docker login
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build and push cloud image
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        file: docker-build/Dockerfile.cloud
+        platforms: Linux/${{ matrix.arch }}
+        push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Is there an equivalent of credscan we can run on GitHub?

Reply:
I am not sure if this would catch something like `echo $HUGGINGFACE_TOKEN > $HOME/.huggingface/token` in the Dockerfile (where `$HUGGINGFACE_TOKEN` is set to `${{ secrets.HUGGINGFACE_TOKEN }}`)

Reply:
Host environment isn't automatically included at build time, so to leak a variable into the image, you'd have to pass the envvar as a build arg, and add a corresponding `ARG` statement in the dockerfile. And yeah @mauwii, you are correct that every PR into Github actions should undergo additional scrutiny. Doesn't look like any secrets are being passed at build time right now though. Also, see [diff](https://github.com/invoke-ai/InvokeAI/pull/1544/files#diff-0b4b60e8ac90c2623e5a6dadf5ad8c401c511d7942c5c1000b76752f64d9c02aR47) - On PRs the pipeline does not attempt to log in to github at all, and on the author's forked branch it will use the PR author's own credentials. And the workflow runs only on `push` (not PR) to `main` or `development`... seems like this covers all bases at the moment?

Reply:
@Kyle0654 Github already does some pretty comprehensive scanning under the hood _([1], [2])_ which is basically the equivalent of CredScan. Given they are owned by MSFT, I wouldn't be surprised if there are some common pieces even (this is me purely speculating :smile: ). But yeah looking into some 3rd party scanners is also never a bad idea IMO.

[1] https://docs.github.com/en/code-security/secret-scanning/about-secret-scanning
[2] https://docs.github.com/en/developers/overview/secret-scanning-partner-program